### PR TITLE
Dev/lf

### PR DESCRIPTION
**Pull request by lvafi**
First PR for code review

### CODE REVIEW

File: src/app/calculator-component/calculator-component.component.ts [hunk 1]
```diff
@@ -1,15 +1,104 @@
...
+  numbers: Array<number> = [];
+  operators: Array<string> = [];
+  result: Number = 0;
+  display: string = "";
+  operator: string = "Add";
+
+  constructor(private calcService: CalculatorService) { }
 
   ngOnInit() {
   }
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.ts]: remove empty ngOnInit

File: src/app/calculator-component/calculator-component.component.ts [hunk 2]
```diff
@@ -1,15 +1,104 @@
...
 
+  enterNumber(number: string) {
+    // if the display already is showing the total result,
+    // and the user enter another digit again, 
+    // the display needs to be cleared to store the new number
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.ts]: If you need a comment for this put into a method with a descriptive method name instead.

File: src/app/calculator-component/calculator-component.component.ts [hunk 3]
```diff
@@ -1,15 +1,104 @@
...
+    try {
+      if (this.state == "enter Number") {
+        this.operateOnSymbol(op);
+      } else if (this.state == "obtain result") {
+        /*
+        if (op == "+" || op == "-" || op == "*" || op == "/") {
+          this.state = "enter Number";
+          this.operateOnSymbol(op);          
+        }
+        */
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.ts]: Do not leave dead code in a comment. remove code or leave a comment of why it's needed in the future.

File: src/app/calculator-component/calculator-component.component.ts [hunk 4]
```diff
@@ -1,15 +1,104 @@
...
+    }
+  }
+
+  private operateOnSymbol(op) {
+    // detect to see if operator is a backspace
+    // this will remove the last digit(or decimal) entered in the display
+    if (op == 'backspace') {
+      this.display = this.display.substring(0, this.display.length - 1);
+      return;
+    }
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.ts]: move to method

File: src/app/calculator-component/calculator-component.component.ts [hunk 5]
```diff
@@ -1,15 +1,104 @@
...
+    console.log("number is =" + convertedNumber);
+    // clear display
+    this.display = "";
+    // check to see if operation is to obtain the answer
+    if (op == "=") {
+      // call service to obtain result of requested operation
+      this.state = "obtain result"
+      this.display = this.getresult(this.operators).toString();
+    }
+  }
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.ts]: this method is too long, consider breaking it up into smaller methods, each method should do only one thing.

File: src/app/calculator-component/calculator-component.component.ts [hunk 6]
```diff
@@ -1,15 +1,104 @@
...
+      convertedNumber = 0;
+    }
+    else {
+      // copy the display to number
+      convertedNumber = parseFloat(this.display);
+    }
+
+    this.numbers.push(convertedNumber);
+
+    console.log("number is =" + convertedNumber);
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.ts]: remove debugging

File: src/app/calculator.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+//import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
+
+//import { CalculatorComponentComponent } from './calculator-component/calculator-component.component';
```
Comment (deefactorial) [inline src/app/calculator.service.spec.ts]: remove unused code.

File: src/app/calculator.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+// services
+import { CalculatorService } from './calculator.service';
+
+describe('CalculatorService', () => {
+  beforeEach(() => TestBed.configureTestingModule({
+    declarations: [
+      //CalculatorComponentComponent
```
Comment (deefactorial) [inline src/app/calculator.service.spec.ts]: remove unused code

File: src/app/calculator.service.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      var tempNumber = numbers[i];
+      var tempOperator = operators[i-1];
+      switch (tempOperator) {
+        case '+': tempResult = tempResult + tempNumber; break;
+        case '-': tempResult = tempResult - tempNumber; break;
+        case '*': tempResult = tempResult * tempNumber; break;
+        case '/': tempResult = tempResult / tempNumber; break;
+      }
+    }
+    console.log("number is =" + tempResult);
```
Comment (deefactorial) [inline src/app/calculator.service.ts]: remove debugging

File: src/app/calculator.service.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  /**
+   * calculate
+   * returns the result of the array treated with the operators
+   * works with +,-,* and /
+   * @param numbers, an array of numbers to operate on
+   * @param operators an array of operators treat the next number with
+   */
+  calculate(numbers, operators) {
+    // TBD this is mockup function logic
+    // the following logic in production code, will be hidden in the cloud.
```
Comment (deefactorial) [inline src/app/calculator.service.ts]: Can you explain what hidden in the cloud means ?

File: src/tslint.json [hunk 11]
```diff
@@ -1,6 +1,12 @@
 {
-    "extends": "../tslint.json",
+    "extends": ["../tslint.json","angular-tslint-rules"],
     "rules": {
+        "no-unnecessary-class": [
+            true,
+            "allow-constructor-only",
+            "allow-static-only",
+            "allow-empty-class"
+        ],
```
Comment (deefactorial) [inline src/tslint.json:9]: the rules are there for a reason, they are hints on how to improve your code, try returning the rules and fixing the linting errors.

File: src/app/calculator-component/calculator-component.component.html [hunk 12]
```diff
@@ -1,7 +1,82 @@
...
+        <div class="col-sm-1">
+          <button class="btn btn-primary" (click)="enterNumber('7')" >7</button>
+        </div>
+        <div class="col-sm-1">
+          <button class="btn btn-primary" (click)="enterNumber('7')" >7</button>
+        </div>
+        <div class="col-sm-1">
+          <button class="btn btn-primary" (click)="enterNumber('7')" >7</button>
+        </div>
+      </div> -->
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.html]: remove unused code or comment why it's needed.

File: src/app/calculator-component/calculator-component.component.html [hunk 13]
```diff
@@ -1,7 +1,82 @@
...
+                  </li>
+                </td>
+              </tr>
+          
+          </tbody>
+          </table>
+    </div>
+    <div class="col-sm-2"></div>        
+  </div>
+</div>
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.html]: Can this component be split into smaller parts ?

File: src/app/calculator-component/calculator-component.component.ts [hunk 14]
```diff
@@ -1,15 +1,104 @@
...
+        }
+        */
+      }
+    }
+    catch (ex) {
+
+    }
+  }
+
+  private operateOnSymbol(op) {
```
Comment (deefactorial) [inline src/app/calculator-component/calculator-component.component.ts]: consider moving this logic to the calculator service

**Review by deefactorial — CHANGES_REQUESTED** [review]
There are a couple of improvements to make on the code.

**Comment by lvafi** [comment]
Code review changes in the next PR.

**Pull request closed without merging**